java/arcs/core/storage/database/Database.kt

```diff
@@ -60,6 +60,9 @@ interface Database {
     /** Garbage collection run: will remove unused entities. */
     suspend fun runGarbageCollection()
 
+    /** Returns the current database size, in bytes. */
+    suspend fun getSize(): Long
```

Comment:
we already have a method to check database size in the AndroidSqliteDatabaseManager (see databaseSizeTooLarge function):
context.getDatabasePath(dbName).length()
what is the difference between the two implementations, could we use only one? could you get the size directly from the database manager instead of dbimpl?

Reply:
I think the length() one only works for on-disk databases. I also don't want to explicitly call AndroidSqliteDatabaseManager as it's less generic. 

BTW: I think you might also want to check the in-memory database's size if it grows too large. So maybe we can unify into using this new method I added.

Reply:
I see, if length doesn't work for in-memory, we should switch that function to use your new method instead, nice find!
Have you checked how the two measures compare (maybe run a quick on device test), could this overstimate the size if the pages are not full? probably doesn't matter too much anyway.

Reply:
I did some experiments and the results are quite close. So I think it's OK to replace the existing one with this new one.